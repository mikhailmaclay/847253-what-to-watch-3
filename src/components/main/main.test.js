// Libraries
import React from 'react';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
// Components
import Main from './main';

const specialMovie = {
  id: 1,
  name: `The Grand Budapest Hotel`,
  genre: `Comedy`,
  releaseDate: `2014`,
  images: [``],
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  background: `img/bg-the-grand-budapest-hotel.jpg`
};

describe(`<Main/> should render correctly`, () => {
  const props = {
    genres: [`Comedy`],
    specialMovie,
    movies: [],
    currentGenre: null,
    onGenreChange: () => {}
  };

  it(`should render correctly`, () => {
    const result = renderer.create(
        <BrowserRouter>
          <Main {...props}/>
        </BrowserRouter>
    ).toJSON();

    expect(result).toMatchSnapshot();
  });
});
