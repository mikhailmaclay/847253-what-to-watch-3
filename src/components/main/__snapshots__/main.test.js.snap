// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main/> Should render correctly 1`] = `
Array [
  <div
    className="movie-card__bg"
  />,
  <h1
    className="visually-hidden"
  >
    WTW
  </h1>,
  <header>
    <div
      className="logo"
    >
      <a
        className="logo__link "
        href={null}
        onClick={null}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div
      className="user-block"
    >
      <a
        className="user-block__link"
        href="/sign-in"
        onClick={[Function]}
      >
        Sign in
      </a>
    </div>
  </header>,
  <div
    className="page-content"
  >
    <section
      className="catalog"
    >
      <h2
        className="visually-hidden"
      >
        Catalog
      </h2>
      <div
        className="catalog__movie-list"
      >
        <article
          className="small-movie-card catalog__movie-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__image"
            href="/films/1"
            onClick={[Function]}
          >
            <img
              alt="Shutter Island"
              height="175"
              src="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/shutter-island.jpg"
              width="280"
            />
          </a>
          <h3
            className="small-movie-card__title"
          >
            <span
              className="small-movie-card__link"
            >
              Shutter Island
            </span>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movie-card"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <a
            className="small-movie-card__image"
            href="/films/2"
            onClick={[Function]}
          >
            <img
              alt="War of the Worlds"
              height="175"
              src="https://htmlacademy-react-3.appspot.com/wtw/static/film/preview/war-of-the-worlds.jpg"
              width="280"
            />
          </a>
          <h3
            className="small-movie-card__title"
          >
            <span
              className="small-movie-card__link"
            >
              War of the Worlds
            </span>
          </h3>
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href={null}
          onClick={null}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
